### PR TITLE
[kotlin compiler][update] 1.4.0-dev-2781 

### PR DESCRIPTION
**Pull request by vvlevchenko**
* 220bf6d62f7 - (tag: build-1.4.0-dev-2781) [NI] Update test data (vor 14 Stunden) <Mikhail Zarechenskiy>
* 3461effd477 - (tag: build-1.4.0-dev-2778) KT-33372 Remove renaming file reference to the contents of the file (vor 21 Stunden) <Roman Golyshev>
* 50c477bee10 - (tag: build-1.4.0-dev-2775, origin/rr/pr3080) Update IR text test expectations for FIR (vor 2 Tagen) <Steven Schäfer>
* de33b83bb3a - Update asFunKSmall.kt (vor 2 Tagen) <Kristoffer Andersen>
* 577250d73f4 - [FIR] Fix FIR2IR Types on SetVariable Expressions (vor 2 Tagen) <Kristoffer Andersen>
* 4dc6583f313 - Update asFunKSmall.kt (vor 2 Tagen) <Kristoffer Andersen>
* 4f36e2ccfcb - [FIR] Fix FIR2IR Types on SetVariable Expressions (vor 2 Tagen) <Kristoffer Andersen>
* 1e7f8ea0372 - PSI2IR test for coercion to unit in lambda (vor 2 Tagen) <Dmitry Petrov>
* 5760c0be9cd - JVM IR: Avoid redundant coercions in ExpressionCodegen (vor 2 Tagen) <Steven Schäfer>
* 2b2ae8a5f1c - IR: Optimize implicit integer coercions on constants. (vor 2 Tagen) <Steven Schäfer>
* bc2b96f6344 - IR: Fix return types in IrConstImpl (vor 2 Tagen) <Steven Schäfer>
* 2677fa74729 - (tag: build-1.4.0-dev-2772) .gradle.kts: more clear tooltip for out of project scripts notification (vor 2 Tagen) <Sergey Rostov>
* 52bf630f6c0 - Add ApplicationUtils.as40 as it does not have CancellationCheck (vor 2 Tagen) <Vladimir Dolzhenko>
* ba5a335e95a - (tag: build-1.4.0-dev-2771) Add missed create in usage of new annotation highlighting API (vor 2 Tagen) <Vladimir Dolzhenko>
* 3407904e682 - (tag: build-1.4.0-dev-2757) [JS IR] Add todo (vor 3 Tagen) <Roman Artemev>
* e57d34dd9ef - Make fix test: (vor 3 Tagen) <Roman Artemev>
* 46491c12eec - [KLIB] Promote Klib ABI version (vor 3 Tagen) <Roman Artemev>
* a411549e38b - [KLIB] Fix zipped layout (vor 3 Tagen) <Roman Artemev>
* 5004bb36360 - [Psi2Ir] Fix generation of hashcode for generic inline class (vor 3 Tagen) <Roman Artemev>
* 6a37955a360 - [KLIB] Implement new linker based on IdSignature (vor 3 Tagen) <Roman Artemev>
* 31d73c5d795 - [Psi2Ir] Move to new linkage scheme based on IdSignature (vor 3 Tagen) <Roman Artemev>
* df7b7cf61aa - [K/N] Provide original class id for forward-declared declarataion (vor 3 Tagen) <Roman Artemev>
* e81a7f10c13 - [KLIB] Implement special encoding for Klib data (vor 3 Tagen) <Roman Artemev>
* 76881978414 - [JS IR] Fix access to coroutine empty context for suspend main (vor 3 Tagen) <Roman Artemev>
* cef9ed1dae6 - [IR] Implement new IrSymbol API (vor 3 Tagen) <Roman Artemev>
* bb04eae93e7 - [IR] Implement IdSignature API (vor 3 Tagen) <Roman Artemev>
* 45f036548b9 - [Psi2Ir] Make IrEnumEntry corresponding class always privite (vor 3 Tagen) <Roman Artemev>
* d51b14fe3cf - [IR] Fix classic mangler expect checker (vor 3 Tagen) <Roman Artemev>
* 0254734bb53 - [IR] Refactored Kotlin Mangler API (vor 3 Tagen) <Roman Artemev>
* ba25b0faafb - (tag: build-1.4.0-dev-2754) [NI] Report unsafe invoke properly for flexible types (vor 3 Tagen) <Pavel Kirpichenkov>
* b1618390929 - [NI] Report unsafe implicit invoke accordingly to OI (vor 3 Tagen) <Pavel Kirpichenkov>
* 1a2d28d25b5 - (tag: build-1.4.0-dev-2752) [FIR-TEST] Add test with callable reference on function with default argument (vor 3 Tagen) <Dmitriy Novozhilov>
* 3790f31d802 - [FIR] Support `FirVarargArgumentsExpression` in html dump (vor 3 Tagen) <Dmitriy Novozhilov>
* e696d1d06ae - [FIR] Improve mapping arguments to parameters (vor 3 Tagen) <Dmitriy Novozhilov>
* 2e597a3a328 - (tag: build-1.4.0-dev-2751) Use new annotation highlighting API (vor 3 Tagen) <Vladimir Dolzhenko>
* 2a55560f3b3 - [FIR TEST] Add test on qualifier resolve priority (corner case) (vor 3 Tagen) <Mikhail Glukhikh>
* abfe566255e - [FIR] Add object support as implicit invoke receiver (vor 3 Tagen) <Mikhail Glukhikh>
* a8a696c7e1e - [FIR TEST] Extend test on invoke priorities (vor 3 Tagen) <Mikhail Glukhikh>
* 79060e7f40d - (tag: build-1.4.0-dev-2747) IR: handle conflicts in resolveFakeOverride() (vor 3 Tagen) <Georgy Bronnikov>
* c71e87068a9 - JVM_IR: in InheritedDefaultMethodsOnClassesLowering, skip bridges when resolving overrides (vor 3 Tagen) <Georgy Bronnikov>
* 829e5908d0d - (tag: build-1.4.0-dev-2746) Allow Native `@Throws` on override when the same list is specified (vor 3 Tagen) <Svyatoslav Scherbina>
* 3fcdf6c78c9 - (tag: build-1.4.0-dev-2745) Update dukat dependency to 0.0.28 (vor 3 Tagen) <Shagen Ogandzhanian>
* 64f13ab3ae2 - (tag: build-1.4.0-dev-2744) [Gradle, JS] Migrate on strings (vor 3 Tagen) <Ilya Goncharov>
* 9b462b76d4e - [Gradle, JS] Add all possible values of library target as enums (vor 3 Tagen) <Ilya Goncharov>
* fbabbfab44e - [Gradle, JS] Add webpack output (vor 3 Tagen) <Ilya Goncharov>
* dc77df10838 - (tag: build-1.4.0-dev-2734) GradleScriptInputsWatcher: move long running operation out of EDT (vor 3 Tagen) <Natalia Selezneva>
* 2d15914d205 - (tag: build-1.4.0-dev-2732) Enable new inference in IDE by default (vor 3 Tagen) <Mikhail Zarechenskiy>
* bdf1441c803 - (tag: build-1.4.0-dev-2730) Remove special casing for Unit in kotlinx.serialization (vor 3 Tagen) <Leonid Startsev>
* 59cdf3c52e2 - (tag: build-1.4.0-dev-2729) Update bytecode text tests in JVM_IR (vor 3 Tagen) <Dmitry Petrov>
* db91b520a5f - (tag: build-1.4.0-dev-2728) [minor] Fix test source - avoid warning about uused main args (vor 3 Tagen) <Ilya Chernikov>
* 63568079978 - Reapply "Only create descriptors for candidates with lambda args" (vor 3 Tagen) <Ilya Chernikov>
* 238bfe37714 - (tag: build-1.4.0-dev-2721) Fix compilation (vor 3 Tagen) <Dmitry Gridin>
* 1c945430bc7 - (tag: build-1.4.0-dev-2717) Fix class loading deadlock in Error diagnostic classes (vor 3 Tagen) <Yan Zhulanow>
* 2f34bc88dce - Debugger: Add test for KT-14869 (vor 3 Tagen) <Yan Zhulanow>
* ea7ffa70b33 - Debugger: Add test for KT-34905 (vor 3 Tagen) <Yan Zhulanow>
* 6690f946d6b - Debugger: Add test for KT-14296 (vor 3 Tagen) <Yan Zhulanow>
* 6f425ee9515 - Debugger: Add test for KT-15652 (vor 3 Tagen) <Yan Zhulanow>
* dd33640238e - Debugger: Skip inlined library frames on step over (KT-12016) (vor 3 Tagen) <Yan Zhulanow>
* 7490c229ac8 - EA-219152 (partially): Support KClass 'unboxing' in debugger (vor 3 Tagen) <Yan Zhulanow>
* a7071344f54 - EA-219152 (partially): Provide correct PropertyDescriptor for _field (vor 3 Tagen) <Yan Zhulanow>
* 9c889aab173 - EA-213309: Eliminate "Invalid method id" exception (vor 3 Tagen) <Yan Zhulanow>
* f6b62f22998 - Minor: extract wrapEvaluateException() (vor 3 Tagen) <Yan Zhulanow>
* 3027fab2388 - EA-218160: Force-box only assignment l-values (vor 3 Tagen) <Yan Zhulanow>
* d539a870edb - EA-214668, EA-209980: Use StackFrameProxyImpl with safe implementation of thisObject() (vor 3 Tagen) <Yan Zhulanow>
* fabbfdc7f81 - EA-214654: Add missing read action (vor 3 Tagen) <Yan Zhulanow>
* dc71ab5bb37 - EA-211010: Fix receiver parameter name for lambdas passed as value arguments (vor 3 Tagen) <Yan Zhulanow>
* b1414fa4770 - EA-214662: Support property accessors in UastLightIdentifier (vor 3 Tagen) <Yan Zhulanow>
* be99f44c4f8 - EA-209623: Check library disposed state (vor 3 Tagen) <Yan Zhulanow>
* eb3685d6ace - EA-218902: Check PsiElement for validity (vor 3 Tagen) <Yan Zhulanow>
* 615d24bbd2b - Minor: Rename KotlinStepOverInlineFilter to KotlinStepOverFilter (vor 3 Tagen) <Yan Zhulanow>
* f4337eede12 - Debugger, BE: Support lambda breakpoints for lambdas passed to '@InlineOnly' functions (vor 3 Tagen) <Yan Zhulanow>
* f129ab55c59 - Debugger: Rewrite step out action (vor 3 Tagen) <Yan Zhulanow>
* cc2fe6b0c60 - Debugger, BE: Generate original line number after inlining if call is used in an if condition (vor 3 Tagen) <Yan Zhulanow>
* f115bde682d - Debugger, BE: Provide Java-like stepping for 'for' loops (vor 3 Tagen) <Yan Zhulanow>
* 8b177180869 - Debugger, BE: Make the NOP instruction optimizer save the first NOP, not the last (vor 3 Tagen) <Yan Zhulanow>
* 9cba72d5dd3 - Debugger: Remove flaky sleep-based stepping tests (vor 3 Tagen) <Yan Zhulanow>
* 01a6e7abf31 - Debugger, eval4j: Remove eager class loading in findField() (KT-35354) (vor 3 Tagen) <Yan Zhulanow>
* 8f29f8bc9da - Debugger: Rewrite step over action (KT-14296) (vor 3 Tagen) <Yan Zhulanow>
* b7449c2d6e4 - Debugger, minor: Move Action to its own file, rename to KotlinStepAction (vor 3 Tagen) <Yan Zhulanow>
* dd4a8d01c9c - Debugger: Remove obsolete hacks needed for old Android dex (vor 3 Tagen) <Yan Zhulanow>
* dd743f05c1a - Debugger: Remove/change tests for the deprecated Android dex (vor 3 Tagen) <Yan Zhulanow>
* f912602ddcf - Debugger, minor: Move KotlinSourcePosition to top level (vor 3 Tagen) <Yan Zhulanow>
* 2df646e2738 - Debugger, minor: Remove unneeded nullability dance (vor 3 Tagen) <Yan Zhulanow>
* 1a3553cdb76 - Debugger: Simplify method filters, use smart pointer in KotlinLambdaMethodFilter (vor 3 Tagen) <Yan Zhulanow>
* c6262b5ff20 - Debugger, minor: Fix element equivalence check in basic step filter (vor 3 Tagen) <Yan Zhulanow>
* fc09534464e - Debugger, minor: move step filters to the 'filter' package (vor 3 Tagen) <Yan Zhulanow>
* d53c554c8c9 - Debugger, minor: Add missing 'const' modifier (vor 3 Tagen) <Yan Zhulanow>
* 6d606e5291e - Debugger, minor: Move createStepOverCommandWithCustomFilter() to DebuggerSteppingFilter (vor 3 Tagen) <Yan Zhulanow>
* 26371ea023b - Debugger, minor: Update getActiveFilters() implementation, use prepend() (vor 3 Tagen) <Yan Zhulanow>
* c196ce5f096 - Pill: Pack embedded dependencies to artifacts (vor 3 Tagen) <Yan Zhulanow>
* a2be789d37f - Pill: Move out Pill tasks from buildSrc (vor 3 Tagen) <Yan Zhulanow>
* 6f1b3e0f17b - Pill: Fix pre-built run configurations (vor 3 Tagen) <Yan Zhulanow>
* 73813aef235 - Pill: Support all Gradle source sets, not just main/test (vor 3 Tagen) <Yan Zhulanow>
* 3acf7a46791 - (tag: build-1.4.0-dev-2716) TCServiceMessageOutputStreamHandlerTest: fix tests (vor 3 Tagen) <nataliya.valtman>
* 389dcff1255 - (tag: build-1.4.0-dev-2712) [FIR] Discard argument with spread in place of non-vararg parameter (vor 3 Tagen) <Dmitriy Novozhilov>
* 50ac8eb95be - [FIR] Add smartcast info to argument of invoke of function with receiver (vor 3 Tagen) <Dmitriy Novozhilov>
* 4fbed9b3633 - [FIR] Add builder for `FirExpressionWithSmartcast` (vor 3 Tagen) <Dmitriy Novozhilov>
* 1da036ac98a - [FIR-TEST] Split outdated dfa test to separate tests (vor 3 Tagen) <Dmitriy Novozhilov>
* 1dd74177464 - [FIR] Fix configuring label name for lambdas in light tree builder (vor 3 Tagen) <Dmitriy Novozhilov>
* 6b378cab29e - [FIR] Don't render lambda bodies in arguments of calls in cfg dump (vor 3 Tagen) <Dmitriy Novozhilov>
* 28c95b1108d - [FIR] Don't use `psi` for rendering control flow graph (vor 3 Tagen) <Dmitriy Novozhilov>
* a4b53b4a20d - [FIR] Fix passing flow from contracts with form `returns(false) ...` (vor 3 Tagen) <Dmitriy Novozhilov>
* 643a4b9c3b8 - [FIR] Support `makeDefinitelyNotNull` for intersection types (vor 3 Tagen) <Dmitriy Novozhilov>
* b71e4ebb3fb - [FIR-TEST] Add `!DUMP_CFG` directive. Get rid of cfg diagnostic tests (vor 3 Tagen) <Dmitriy Novozhilov>
* f495c4e2156 - [FIR-TEST] Move stdlib resolve tests out from `resolve` directory (vor 3 Tagen) <Dmitriy Novozhilov>
* 90021f3ecee - [FIR-TEST] Reorganize tests for data flow analysis (vor 3 Tagen) <Dmitriy Novozhilov>
* 89fa3572eca - [FIR-TEST] Move tests for solved issues from `problems` directory (vor 3 Tagen) <Dmitriy Novozhilov>
* 0c9720f20ce - (tag: build-1.4.0-dev-2709) ReplaceGetOrSetInspection: do not report for 'get' with spread operator (vor 3 Tagen) <Toshiaki Kameyama>
* b782e8f0f01 - (tag: build-1.4.0-dev-2705) Add IR equivalent of AbstractBytecodeListingTest and muted failures. (vor 4 Tagen) <Mark Punzalan>
* 06b6477d044 - [JVM IR] Do not add nullability annotation for functions that return platform types. (vor 4 Tagen) <Mark Punzalan>
* 0c984c5e62f - PSI2IR tests for b29993739e85dbceb02d04a964741b1410a568f8 (vor 4 Tagen) <Dmitry Petrov>
* 5a49ccac760 - JVM_IR: Relax bailout condition for SAM conversion generation. (vor 4 Tagen) <Jinseong Jeon>
* f782ea075ba - (tag: build-1.4.0-dev-2703) [JVM IR] ProperVisibilityForCompanionObjectInstanceField (vor 4 Tagen) <Kristoffer Andersen>
* a67d97bdf8a - (tag: build-1.4.0-dev-2686) Remove redundant fake facade classes (vor 4 Tagen) <Igor Yakovlev>
* 5d6d287faa9 - (tag: build-1.4.0-dev-2682) [NI] Refactor UnstableSmartCast diagnostic (vor 4 Tagen) <Pavel Kirpichenkov>
* 15d744c3da4 - [NI] Continue resolution after unstable smart cast on receiver (vor 4 Tagen) <Pavel Kirpichenkov>
* 4997c9535cb - Mute test, affected by changes in unstable smart casts (vor 4 Tagen) <Pavel Kirpichenkov>
* e40ba73950d - [NI] Add resolution applicability for resolved candidates with error (vor 4 Tagen) <Pavel Kirpichenkov>
* 4fa87d2caa8 - (tag: build-1.4.0-dev-2681) [Gradle, JS] Compilations with js type because of compatibility (vor 4 Tagen) <Ilya Goncharov>
* 76f8109ff60 - (tag: build-1.4.0-dev-2675) JVM IR: Implement LookupLocation in IrSourceCompilerForInline (vor 4 Tagen) <Steven Schäfer>
* 3278451b073 - (tag: build-1.4.0-dev-2673) JVM IR: do not generate certain special bridges as final (vor 4 Tagen) <Alexander Udalov>
* df046683cc1 - (tag: build-1.4.0-dev-2671) Revert "[NI] Report unsafe implicit invoke accordingly to OI" (vor 4 Tagen) <Pavel Kirpichenkov>
* bf9d4f065b8 - (tag: build-1.4.0-dev-2669) Uast: eliminating the `kotlin.uast.multiresolve.enabled`, it is always `true` for a long time (vor 4 Tagen) <Nicolay Mitropolsky>
* e4b7c391105 - Uast: forcing `kotlin.uast.force.uinjectionhost=true` always (vor 4 Tagen) <Nicolay Mitropolsky>
* ee48580e06f - (tag: build-1.4.0-dev-2664) [NI] Update test data (vor 4 Tagen) <Mikhail Zarechenskiy>
* 531a63bd190 - Show experimental features panel only for non-stable plugin versions (vor 4 Tagen) <Ilya Kirillov>
* b045adf83a8 - (tag: build-1.4.0-dev-2661) [NI] Report unsafe implicit invoke accordingly to OI (vor 4 Tagen) <Pavel Kirpichenkov>
* 9d511802021 - Guard resolve calls with runWithCancellationCheck (vor 4 Tagen) <Vladimir Dolzhenko>
* 74becc7e96d - (tag: build-1.4.0-dev-2659) Fix compilation of idea module for old inference (vor 4 Tagen) <Mikhail Zarechenskiy>
* 918895b88ff - (tag: build-1.4.0-dev-2655) Avoid using quite new ScratchFileService API (vor 4 Tagen) <Vladimir Dolzhenko>
* aa0c2486c35 - (tag: build-1.4.0-dev-2654) Fix GenerateTests after performance tests modularization (vor 4 Tagen) <Vladimir Dolzhenko>
* b96109f23f4 - (tag: build-1.4.0-dev-2653) Add diagnostic code to the ScriptDiagnostic (vor 4 Tagen) <Ilya Chernikov>
* 77eacae958d - [minor] drop unused dependencies (vor 4 Tagen) <Ilya Chernikov>
* 49d092cae80 - [minor] add incomplete test to the repl tests (vor 4 Tagen) <Ilya Chernikov>
* 0761a17e2fb - Fix embedding and relocation in main-kts (vor 4 Tagen) <Ilya Chernikov>
* 514282fb0a6 - Use correct platform classloader on script evaluation in cli compiler (vor 4 Tagen) <Ilya Chernikov>
* 489c4ec5ecd - Fix "progressive" warnings to fix compilation under progressive mode (vor 4 Tagen) <Mikhail Zarechenskiy>
* 3d28945ecbe - (tag: build-1.4.0-dev-2644) [NI] Disable feature explicitly as it's required in test (vor 4 Tagen) <Mikhail Zarechenskiy>
* 7526162f6f6 - (tag: build-1.4.0-dev-2642) Mute one inline codegen test for NI (vor 4 Tagen) <Mikhail Zarechenskiy>
* 3702d31cbcd - Set latest stable version for diagnostic tests with unsigned types (vor 4 Tagen) <Mikhail Zarechenskiy>
* 866f1881205 - (tag: build-1.4.0-dev-2640) Report JVM signature clashes from JVM_IR (vor 4 Tagen) <Dmitry Petrov>
* c1db9b6d407 - (tag: build-1.4.0-dev-2638) [NI] Fix incremental compilation for SAM adapters (vor 4 Tagen) <Mikhail Zarechenskiy>
* 1ba0870ae35 - [NI-MIGRATE] Update IR text tests (vor 4 Tagen) <Mikhail Zarechenskiy>
* abc5eb47407 - [NI-MIGRATE-BAD] Update problematic/questionable tests (vor 4 Tagen) <Mikhail Zarechenskiy>
* c2fc633ad6e - [NI-MIGRATE] Update test about signature enhancements (vor 4 Tagen) <Mikhail Zarechenskiy>
* 319a38bd5cb - [NI-MIGRATE] Update diagnostic messages (vor 4 Tagen) <Mikhail Zarechenskiy>
* cd0012c5274 - [NI-MIGRATE] Update CLI tests (vor 4 Tagen) <Mikhail Zarechenskiy>
* ca57d6cd169 - [NI-MIGRATE] Update pseudocode tests for NI (vor 4 Tagen) <Mikhail Zarechenskiy>
* a0071885bff - [NI-MIGRATE] Update type binding tests for NI (vor 4 Tagen) <Mikhail Zarechenskiy>
* 687c96060fb - [NI-MIGRATE] Update test data for resolve tests after enabling NI (vor 4 Tagen) <Mikhail Zarechenskiy>
* 17df92e999d - Fix case in stdlib where more specific type lead to error (vor 4 Tagen) <Mikhail Zarechenskiy>
* 599ff4711e5 - Temporary fix code before proper support of UnsafeVariance in NI (vor 4 Tagen) <Mikhail Zarechenskiy>
* c64577b7350 - Temporary fixes for Kotlin project after enabling NI (vor 4 Tagen) <Mikhail Zarechenskiy>
* cdf5323f8cc - Disable NewInference for stdlib tests (vor 4 Tagen) <Mikhail Zarechenskiy>
* 2a4caf6dc85 - Enable new inference in the compiler (vor 4 Tagen) <Mikhail Zarechenskiy>
* 8da05ed5976 - (tag: build-1.4.0-dev-2636) RemoveReturnLabelFix: introduce `labelName` parameter (vor 4 Tagen) <Dmitry Gridin>
* 7c65bb2e3f7 - AbstractDiagnosticBasedMigrationInspection: cleanup code (vor 4 Tagen) <Dmitry Gridin>
* b98bdca849f - RemoveAnnotationFix: inline default argument (vor 4 Tagen) <Dmitry Gridin>
* f8a21340d05 - ProhibitRepeatedUseSiteTargetAnnotationsMigrationInspection: add test (vor 4 Tagen) <Dmitry Gridin>
* 09e6ffdb0a4 - AbstractDiagnosticBasedMigrationInspection: make lazy initialization (vor 4 Tagen) <Dmitry Gridin>
* cfc2c0fb20c - (tag: build-1.4.0-dev-2633) Changes after rebasing on master (vor 4 Tagen) <Valentin Kipyatkov>
* 4f5d1214789 - Minor code improvements (vor 4 Tagen) <Valentin Kipyatkov>
* e7c72637a17 - Removed misleading comments from test data (vor 4 Tagen) <Valentin Kipyatkov>
* ae578e0a5d7 - Copyright comments (vor 4 Tagen) <Valentin Kipyatkov>
* ee6d6adf707 - processAllExactUsages() only for KtDeclaration (vor 4 Tagen) <Valentin Kipyatkov>
* c33a4c5c5b2 - Code formatting (vor 4 Tagen) <Valentin Kipyatkov>
* 3ccb8fa3a82 - Workaround for compiler crash (vor 4 Tagen) <Valentin Kipyatkov>
* e71bd4e4a4e - No need to intersect with element's useScope - it's performed by reference search by default (vor 4 Tagen) <Valentin Kipyatkov>
* f43626f57f5 - Refactoring (vor 4 Tagen) <Valentin Kipyatkov>
* 826c5dd92e3 - Minor (vor 4 Tagen) <Valentin Kipyatkov>
* 3897f51f8ea - Refactoring code (vor 4 Tagen) <Valentin Kipyatkov>
* f60d24c6217 - Refactoring (vor 4 Tagen) <Valentin Kipyatkov>
* 1b5b0d2839b - Minor refactoring (vor 4 Tagen) <Valentin Kipyatkov>
* af32cad1aeb - Minor (vor 4 Tagen) <Valentin Kipyatkov>
* fd7331bbc0a - Refactoring (vor 4 Tagen) <Valentin Kipyatkov>
* 8008185b77b - Renamed mistyping in function name (vor 4 Tagen) <Valentin Kipyatkov>
* 8f7648c60ab - Reordered methods (vor 4 Tagen) <Valentin Kipyatkov>
* ff398806771 - Refactoring - converting extensions to normal functions (vor 4 Tagen) <Valentin Kipyatkov>
* b1cd56ba918 - We must check super property usages also when analyzing inflow (vor 4 Tagen) <Valentin Kipyatkov>
* 612fd6d1bfd - Refactoring to split big file (vor 4 Tagen) <Valentin Kipyatkov>
* ff516cac73c - Do not include overriders when analyzing parameter usages unless invoked on parameter (vor 4 Tagen) <Valentin Kipyatkov>
* e76eaa4b5b9 - Refactored code to avoid duplication (vor 4 Tagen) <Valentin Kipyatkov>
* b406d85ca32 - Analyze Data Flow to Here: Java super-methods supported (vor 4 Tagen) <Valentin Kipyatkov>
* cfa1d91aa47 - Analyze Data Flow to Here to work with virtual methods (vor 4 Tagen) <Valentin Kipyatkov>
* c34286a3278 - More usable rendering of usage context in Analyze Data Flow to/from Here (vor 4 Tagen) <Valentin Kipyatkov>

### CODE REVIEW

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/serialization/KonanIrModuleSerializer.kt [hunk 1]
```diff
@@ -1,49 +1,40 @@
...
      */
-    override fun tryComputeBackendSpecificUniqId(declaration: IrDeclaration): UniqId? {
-        return if (declaration.descriptor.module.isFromInteropLibrary()) {
-            // Property accessor doesn't provide UniqId so we need to get it from the property itself.
-            UniqId(declaration.descriptor.propertyIfAccessor.getUniqId() ?: error("No uniq id found for ${declaration.descriptor}"))
-        } else {
-            null
-        }
-    }
+//    override fun tryComputeBackendSpecificUniqId(declaration: IrDeclaration): UniqId? {
```
Comment (olonho) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/serialization/KonanIrModuleSerializer.kt]: We usually do not leave commented out code. Why it cannot be just removed?

File: backend.native/tests/build.gradle [hunk 2]
```diff
@@ -3381,7 +3381,7 @@ if (PlatformInfo.isAppleTarget(project)) {
     createInterop("objcGh3343") {
         it.defFile 'framework/gh3343/objclib.def'
         it.headers "$projectDir/framework/gh3343/objclib.h"
-        it.linkerOpts "-L$buildDir"
+        it.linkerOpts "-L$buildDir",  "-lobjcgh3343"
```
Comment (olonho) [inline backend.native/tests/build.gradle]: It it needed, think it was recently intentionally removed.

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/serialization/KonanIrModuleSerializer.kt (hunk 1 above)
Comment (vvlevchenko) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/serialization/KonanIrModuleSerializer.kt]: I've thought mirror contains reviewed commits :/
